pkg/controller/deployment/deployment_controller.go

```diff
@@ -1412,6 +1412,15 @@ func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
 	// If there is no new replica set matching this deployment and the deployment isn't paused
 	// then there is a new rollout that waits to happen
 	if newRS == nil && !d.Spec.Paused {
+		// Update all active replicas sets to the new deployment size. setReplicasAnnotations makes
+		// sure that we will update only replica sets that don't have the current size of the deployment.
+		for _, rs := range controller.FilterActiveReplicaSets(oldRSs) {
+			if updated := setReplicasAnnotations(rs, d.Spec.Replicas, d.Spec.Replicas+maxSurge(*d)); updated {
```

Comment:
Yes, but if we end up here, we are deferring to the strategy. This is a special case where you are changing the replica size and the pod template. The strategy will scale up the new replica set or scale down an old replica set but it won't update the rest of the replica sets to the new desired size. Next sync of the deployment should find that all active replica sets have the desired size and continue the rollout.